### PR TITLE
GEODE-2269 update to allow region entries non null empty key such as …

### PR DESCRIPTION
**Pull request by ggreen**
update to allow region entries non null empty key such as zero length strings to be removed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,12 @@ build/
 .idea/
 build-eclipse/
 /tags
+bin
+runtime
```
Comment (metatype) [inline .gitignore]: Thanks for the contribution!  Did you mean to include the `.gitignore` changes in this PR?

**Comment by ggreen** [comment]
Hello

I did, but feel free ignore the .gitignore if it causes any issues

**Comment by upthewaterspout** [comment]
+1 I'll merge the non .gitignore changes. I don't think we should be ignoring directories named bin, we actually have some code checked in those directories.

I suspect maybe you are using eclipse? The default eclipse behavior is to create an output directory called bin. If so, you should use ./gradlew eclipse and then just import the generated eclipse project into your workspace. Not only will that fix the output directory but it will setup your classpath for you!

**Comment by ggreen** [comment]
Sounds good. Yes please ignore the .gitignore

Thank you Dan

**Pull request closed without merging**